tests/numpy/test_arithmetic.py

```diff
@@ -145,7 +145,69 @@ def test_inner_outer(nps_app_inst):
     assert np.allclose(np.outer(A, B), nps.outer(nps_A, nps_B).get())
 
 
-# TODO(hme): Add broadcast tests.
+def test_broadcast_error(nps_app_inst):
```

Comment:
We're testing for valid tensordot shapes, which is different than broadcasting. The broadcasting tests ought to be for element-wise operations (and matmul for stacked matrices I believe). Can we revive the broadcast tests TODO, and rename this test to something like `test_tensordot_shape_errors` (and something consistent with the `check_` functions below)?

If you want to add broadcast tests, would be best to add those as a separate PR.